torchvision/prototype/datasets/_builtin/imagenet.py

```diff
@@ -108,7 +115,7 @@ def _collate_and_decode_sample(
         return dict(
             path=path,
             image=decoder(buffer) if decoder else buffer,
-            label=torch.tensor(label),
```

Comment:
This is probably an unrelated change for the target of the PR. Could you explain why we keep it as integer instead of a tensor?

Reply:
For the test split `label is None` and so `torch.tensor(label)` would fail. This will properly resolved by #4864 after which we return `Optional[Label]`.